packages/devtools_app_shared/lib/src/service/service_extension_manager.dart

```diff
@@ -324,21 +319,18 @@ final class ServiceExtensionManager with DisposerMixin {
 
         if (isolateRef != _mainIsolate) return false;
 
-        switch (expectedValueType) {
-          case const (bool):
-            final enabled = response.json!['enabled'] == 'true' ? true : false;
+        switch (firstValue) {
+          case bool():
+            final enabled = response.json!['enabled'] == 'true';
             await _maybeRestoreExtension(name, enabled);
-          case const (String):
+          case String():
             final String? value = response.json!['value'];
             await _maybeRestoreExtension(name, value);
-          case const (int):
-          case const (double):
+          case num():
             final value = num.parse(
               response.json![name.substring(name.lastIndexOf('.') + 1)],
             );
             await _maybeRestoreExtension(name, value);
-          default:
-            return true;
```

Comment:
did we intend to remove this return?

Reply:
Yes, it seems redundant , since the switch statement is the last statement in the method. Removing the `default` just drops us down into the final `return true`.